tanner/emulators/template_injection.py

```diff
@@ -0,0 +1,90 @@
+import asyncio
+import logging
+import os
+import docker
+
+from urllib.parse import unquote
+from tanner.utils.php_sandbox_helper import PHPSandboxHelper
+from tanner.utils import patterns
+from tanner.utils import docker_helper
+
+
+class TemplateInjection:
+
+    def __init__(self, loop=None):
+        self._loop = loop if loop is not None else asyncio.get_event_loop()
+        self.logger = logging.getLogger('tanner.template_injection')
+        self.helper = PHPSandboxHelper(self._loop)
+        self.docker_helper = docker_helper.DockerHelper()
+
+    async def get_injection_result_twig(self, payload):
+
+        path_to_vendor = os.getcwd() + '/vendor'
+
+        twig_template = """<?php
+                            require '%s/autoload.php';
+
+                            use Twig\Environment;
+                            use \Twig\Loader\ArrayLoader;
+ 
+                            $name = '%s';
+                            $loader = new ArrayLoader(array('index' => $name,));
+                            $twig = new Environment($loader);
+                            echo $twig->render('index');
+                           ?>""" % (path_to_vendor, payload)
+
+        template_injection_result = await self.helper.get_result(twig_template)
+        return template_injection_result
+
+    def get_injection_result_docker(self, payload):
+        execute_result = None
+
+        client = docker.from_env()
+        docker_image = 'mushorg/template_injection:latest'
```

Comment:
I think you should use `docker_helper`.

Reply:
I tried but the implementation is old and I had to change a lot to make it work. This implementation is described in the latest docs.

https://docker-py.readthedocs.io/en/stable/containers.html#docker.models.containers.ContainerCollection.run

Reply:
I think this is similar to cmd_exec emulator

Reply:
I was difficult to execute complex payloads using old implementation.

Reply:
Then maybe you can update the old implementation so that it can handle complex payloads.

Reply:
Okay! will look into it.